node/src/sub_lib/blockchain_bridge.rs

```diff
@@ -57,10 +67,133 @@ pub struct SetGasPriceMsg {
+    fn batch_wide_timestamp(&self) -> SystemTime;
+    fn send_new_payable_fingerprints_credentials(
+        &self,
+        batch_wide_timestamp: SystemTime,
+        new_pp_fingerprints_sub: &Recipient<NewPendingPayableFingerprints>,
+        payable_attributes: &[(H256, u128)],
+    );
+    fn submit_batch(
+        &self,
+        web3: &Web3<Batch<T>>,
+    ) -> Result<Vec<web3::transports::Result<Value>>, Web3Error>;
+}
+
+#[derive(Debug)]
+pub struct BatchPayableToolsReal<T> {
+    phantom: PhantomData<T>,
+}
+
+impl<T: BatchTransport> Default for BatchPayableToolsReal<T> {
+    fn default() -> Self {
+        Self {
+            phantom: Default::default(),
+        }
+    }
+}
+
+impl<T: BatchTransport + Debug> BatchPayableTools<T> for BatchPayableToolsReal<T> {
+    fn sign_transaction(
+        &self,
+        transaction_params: TransactionParameters,
+        web3: &Web3<Batch<T>>,
+        key: &secp256k1secrets::key::SecretKey,
+    ) -> Result<SignedTransaction, Web3Error> {
+        web3.accounts()
+            .sign_transaction(transaction_params, key)
+            .wait()
+    }
+
+    fn enter_raw_transaction_to_batch(&self, signed_transactions: Bytes, web3: &Web3<Batch<T>>) {
+        let _ = web3.eth().send_raw_transaction(signed_transactions);
+    }
+
+    fn batch_wide_timestamp(&self) -> SystemTime {
+        SystemTime::now()
+    }
+
+    fn send_new_payable_fingerprints_credentials(
+        &self,
+        batch_wide_timestamp: SystemTime,
+        pp_fingerprint_sub: &Recipient<NewPendingPayableFingerprints>,
+        chief_payable_attributes: &[(H256, u128)],
+    ) {
+        pp_fingerprint_sub
+            .try_send(NewPendingPayableFingerprints {
+                batch_wide_timestamp,
+                init_params: chief_payable_attributes.to_vec(),
+            })
+            .expect("Accountant is dead");
+    }
+
+    fn submit_batch(
+        &self,
+        web3: &Web3<Batch<T>>,
+    ) -> Result<Vec<web3::transports::Result<Value>>, Web3Error> {
+        web3.transport().submit_batch().wait()
+    }
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::test_utils::recorder::{make_blockchain_bridge_subs_from, Recorder};
-    use actix::Actor;
+    use crate::blockchain::blockchain_bridge::NewPendingPayableFingerprints;
+    use crate::blockchain::test_utils::{make_tx_hash, TestTransport};
+    use crate::sub_lib::blockchain_bridge::{BatchPayableTools, BatchPayableToolsReal};
+    use crate::test_utils::recorder::{make_blockchain_bridge_subs_from, make_recorder, Recorder};
+    use actix::{Actor, System};
+    use std::time::SystemTime;
+
+    #[test]
+    fn request_new_payable_fingerprints_works() {
+        let (accountant, _, accountant_recording_arc) = make_recorder();
+        let recipient = accountant.start().recipient();
+        let timestamp = SystemTime::now();
+        let chief_attributes_of_payables =
+            vec![(Default::default(), 5), (make_tx_hash(45466), 444444)];
```

Comment:
This `Default::default()` would be easier to understand if it were just `make_tx_hash(0)`.